### PR TITLE
Add Devise

### PR DESCRIPTION
**Pull request by DJ-MrJay**
In this milestone, we:
- Removed <code>current_user</code> method in ApplicationController
- Setup Devise
- Modified the views in Devise for registration and login to match these [wireframes](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/sneak_peek.md)
- Added some more app styling

### CODE REVIEW

File: app/views/devise/sessions/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+      <%= f.label :remember_me %>
+    </div>
+  <% end %>
+
+  <div class="actions">
+    <%= f.submit "Log in" %>
+  </div>
+<% end %>
+
+<%= render "devise/shared/links" %>
```
Comment (elisha2kyakpo1) [inline app/views/devise/sessions/new.html.erb]: - [ ] Kindly modify these views (login and sign up) to match the given wireframe design.

devise


login

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -1,9 +1,11 @@
...
-    @users = User.all
+    # Eager loading posts for all users
+    @users = User.all.includes(:posts)
   end
 
   def show
-    @user = User.find(params[:id])
+    # Eager loading posts, comments, and authors for the specific user
+    @user = User.includes(posts: [{ comments: :author }]).find(params[:id])
   end
```
Comment (elisha2kyakpo1) [inline app/controllers/users_controller.rb]: - [ ] I was able to see the home page (users page) without any login/signup. Users should not type the URL for signup/login to access them. Kindly add a line of code (from devise) that prevents users from seeing pages of the app before login/signup.

**Review by elisha2kyakpo1 — CHANGES_REQUESTED** [review]
Hi 👋 @DJ-MrJay, @serengia, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ The PR title and description are good and professional.
✔️ No linter errors.
✔️ Descriptive readme file.
✔️ Email confirmations👏
👏 Email and password saved and email confirmation.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by brainconnect93 — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi @DJ-MrJay , 👋 👋  

Your project is complete! Great job for following the project requirement🥇and implementing the changes requested by the previous reviewer. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 




## Good Points 👍

- Descriptive Pull Request title & summary. ✔️
- Add login/signup page for user ✔️
- `Devise` is properly implemented ✔️ 
- Good use of Git flow ✔️

## Optional changes

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brainconnect93 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**